packages/nx/src/project-graph/plugins/plugin-pool.ts

```diff
@@ -51,15 +51,20 @@ export function loadRemoteNxPlugin(plugin: PluginConfiguration, root: string) {
 
   // logger.verbose(`[plugin-worker] started worker: ${worker.pid}`);
 
-  return new Promise<RemotePlugin>((res, rej) => {
-    worker.on('message', createWorkerHandler(worker, res, rej));
-    worker.on('exit', createWorkerExitHandler(worker));
-  });
+  return [
+    new Promise<RemotePlugin>((res, rej) => {
+      worker.on('message', createWorkerHandler(worker, res, rej));
+      worker.on('exit', createWorkerExitHandler(worker));
+    }),
+    () => {
+      shutdownPluginWorkers();
```

Comment:
This callback should probably shut down the single plugin that was just loaded. (e.g. shutdownPluginWorker(worker))